### PR TITLE
validate lint name in `clippy_dev`

### PR DESCRIPTION
**Pull request by y21**
This PR adds a little bit of validation to `cargo dev new_lint`. I've had it happen a few times where I wanted to add a new lint, but forgot that lint names cannot contain `-`. If you try to do it anyway, `clippy_dev` will generate illegal syntax (like adding `mod test-lint;` to clippy_lints/src/lib.rs for the module declaration). Maybe having it error out early would be helpful to others too.

changelog: none

### CODE REVIEW

File: clippy_dev/src/main.rs [hunk 1]
```diff
@@ -180,7 +180,14 @@ fn get_clap_config() -> ArgMatches {
                         .short('n')
                         .long("name")
                         .help("Name of the new lint in snake case, ex: fn_too_long")
-                        .required(true),
+                        .required(true)
+                        .value_parser(|name: &str| {
+                            if name.contains('-') {
+                                Err("Lint name cannot contain `-`, use `_` instead.")
```
Comment (flip1995) [inline clippy_dev/src/main.rs:186]: Instead of error, would it make sense to just replace `-` with `_`?

Comment (y21) [reply clippy_dev/src/main.rs:186]: That makes sense to me (assuming that *was* directed at me). I think it's safe to assume that the user probably didn't want the literal character `-` but just meant to have it as a delimiter and replacing it with `_` is fine

**Review by flip1995 — APPROVED** [review]

File: clippy_dev/src/main.rs (hunk 1 above)
Comment (flip1995) [reply clippy_dev/src/main.rs]: Yes, that question was for you :)

File: clippy_dev/src/main.rs [hunk 2]
```diff
@@ -180,7 +181,8 @@ fn get_clap_config() -> ArgMatches {
                         .short('n')
                         .long("name")
                         .help("Name of the new lint in snake case, ex: fn_too_long")
-                        .required(true),
+                        .required(true)
+                        .value_parser(|name: &str| Ok::<_, Infallible>(name.replace("-", "_"))),
```
Comment (flip1995) [inline clippy_dev/src/main.rs:185]:
```suggestion
                        .value_parser(|name: &str| Ok::<_, Infallible>(name.replace('-', "_"))),
```

Clippy is not happy with itself.

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**